src/prefect/agent/docker/agent.py

```diff
@@ -325,24 +321,16 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
         # the 'import prefect' time low
         import docker
 
-        storage = StorageSchema().load(flow_run.flow.storage)
-        if not isinstance(StorageSchema().load(flow_run.flow.storage), Docker):
-            self.logger.error(
-                "Storage for flow run {} is not of type Docker.".format(flow_run.id)
-            )
-            raise ValueError("Unsupported Storage type")
-
+        image = get_flow_image(flow_run=flow_run)
         env_vars = self.populate_env_vars(flow_run=flow_run)
 
-        if not self.no_pull and storage.registry_url:
-            self.logger.info("Pulling image {}...".format(storage.name))
+        if not self.no_pull and len(image.split("/")) > 1:
```

Comment:
What's the reasoning behind this image split?

Reply:
If the full image name doesn't have a repo set (`repo/name:tag`) then don't attempt a pull. It's an extra safeguard which doesn't require to _also_ set `no_pull`